src/prommis/uky/uky_flowsheet.py

```diff
@@ -1020,103 +1607,431 @@ def initialize_system(m):
-        initializer2 = BlockTriangularizationInitializer()
+        initializer_sep = SeparatorInitializer()
+        initializer_mix = MixerInitializer()
+        initializer_leach = LeachingTrainInitializer()
+        initializer_bt = BlockTriangularizationInitializer()
 
         propagate_state(m.fs.liq_feed)
         propagate_state(m.fs.sol_feed)
+        propagate_state(m.fs.org_feed)
+        propagate_state(m.fs.org_feed2)
 
         if stream == m.fs.leach_liquid_feed:
+            print(f"Initializing {stream}")
             initializer_feed.initialize(m.fs.leach_liquid_feed)
         elif stream == m.fs.leach_solid_feed:
+            print(f"Initializing {stream}")
             initializer_feed.initialize(m.fs.leach_solid_feed)
+        elif stream == m.fs.rougher_org_make_up:
+            print(f"Initializing {stream}")
+            initializer_feed.initialize(m.fs.rougher_org_make_up)
+        elif stream == m.fs.acid_feed1:
+            print(f"Initializing {stream}")
+            initializer_feed.initialize(m.fs.acid_feed1)
+        elif stream == m.fs.acid_feed2:
+            print(f"Initializing {stream}")
+            initializer_feed.initialize(m.fs.acid_feed2)
+        elif stream == m.fs.acid_feed3:
+            print(f"Initializing {stream}")
+            initializer_feed.initialize(m.fs.acid_feed3)
+        elif stream == m.fs.cleaner_org_make_up:
+            print(f"Initializing {stream}")
+            initializer_feed.initialize(m.fs.cleaner_org_make_up)
         elif stream == m.fs.leach_filter_cake:
             print(f"Initializing {stream}")
             initializer_product.initialize(m.fs.leach_filter_cake)
+        elif stream == m.fs.load_sep:
+            print(f"Initializing {stream}")
+            initializer_sep.initialize(m.fs.load_sep)
+        elif stream == m.fs.scrub_sep:
+            print(f"Initializing {stream}")
+            initializer_sep.initialize(m.fs.scrub_sep)
         elif stream == m.fs.leach:
             print(f"Initializing {stream}")
             try:
-                initializer1.initialize(m.fs.leach)
+                initializer_leach.initialize(m.fs.leach)
             except:
                 # Fix feed states
                 m.fs.leach.liquid_inlet.flow_vol.fix()
                 m.fs.leach.liquid_inlet.conc_mass_comp.fix()
                 m.fs.leach.solid_inlet.flow_mass.fix()
                 m.fs.leach.solid_inlet.mass_frac_comp.fix()
-                # Re-solve leach unit
+                # Re-solve unit
                 solver = SolverFactory("ipopt")
                 solver.solve(m.fs.leach, tee=True)
                 # Unfix feed states
-                m.fs.leach_liquid_feed.flow_vol.unfix()
+                m.fs.leach.liquid_inlet.flow_vol.unfix()
                 m.fs.leach.liquid_inlet.conc_mass_comp.unfix()
                 m.fs.leach.solid_inlet.flow_mass.unfix()
                 m.fs.leach.solid_inlet.mass_frac_comp.unfix()
         elif stream == m.fs.leach_mixer:
```

Comment:
This is a better way to do this. You could create lists of the units associated with each type of initializer and use `if stream in []`. That way, you need fewer `else` checks.

Reply:
Moved the initializers outside of the function and created the aforementioned if loops.